providers/apis/marketmap/client.go

```diff
@@ -62,6 +62,19 @@ func NewGRPCClient(
 	}, nil
 }
 
+// NewGRPCClientWithConn returns a new GRPC client for MarketMap module with the given connection.
+func NewGRPCClientWithConn(
+	conn *grpc.ClientConn,
+	api config.APIConfig,
+	apiMetrics metrics.APIMetrics,
+) (mmtypes.QueryClient, error) {
+	return &MarketMapClient{
+		QueryClient: mmtypes.NewQueryClient(conn),
+		apiMetrics:  apiMetrics,
+		api:         api,
+	}, nil
+}
```

Comment:
theres some validation stuff done on `ApiConfig` and `ApiMetrics` in `NewGRPCClient` above ; do we need to do that here as well?